### PR TITLE
Use ExecutorService variables to assign ExecutorService Instances

### LINKED ISSUE #9482: An ExecutorService must not be created for each KerberosHttpClient; Don't assign ExecutorService into variable of Executor type
Similarly to #9286, we should catch assignments of `ExecutorService` (within type hierarchy) into variables of `Executor` type.

Two Structural search patterns are needed:
 - "Java" pattern: `$x$ = $y$;`, where the Type of `$x$` is `Executor` and the Type of `$y$` is `ExecutorService`.
 - "Java - Class Member" pattern: `$Type$ $x$ = $y$;`, where the Text of `$Type$` is `Executor` and the Type of `$y$` is `ExecutorService`.

(BTW, there is only "Java" pattern suggested in #9286 and added in #9325; a similar "Java - Class Member" pattern should be added for `ExecutorService` - `ScheduledExecutorService` pair, too.)

This is needed because [`ExecutorService` instances must be shut down explicitly](https://github.com/code-review-checklists/java-concurrency#explicit-shutdown) and `Executor` doesn't allow for this.

Incidentally, the sole violation of this rule in Druid code, in `KerberosHttpClient`:

https://github.com/apache/druid/blob/a6776648112917b72c077ba3ac0cb7f61993a2d0/extensions-core/druid-kerberos/src/main/java/org/apache/druid/security/kerberos/KerberosHttpClient.java#L44-L54

Is actually an instance of another Java concurrency problem: [an `ExecutorService` must not be created for each instance of a short-lived object](https://github.com/code-review-checklists/java-concurrency#reuse-threads) like `KerberosHttpClient`. So the `exec` in `KerberosHttpClient` should properly made static, or cached on the DI level as a `@Named` `ExecutorService`.

### PR DESCRIPTION
**Pull request by shankeerthan-kasilingam**
Fixes #9482

### Description

Added inspection rules to prohibit assigning/initializing `ExecutorService` instances to `Executor` variables.

Changed `Executor` to `ExecutorService` which violates inspection rules.

I looked at the concurrency checklist(Threads and Executors section ), I don't see any violations of the checklist.
Please correct me if anything wrong.

This PR has:
- [ x] been self-reviewed.
   - [ x] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)

### CODE REVIEW

File: .idea/inspectionProfiles/Druid.xml [hunk 1]
```diff
@@ -372,6 +372,16 @@
         <constraint name="l" within="" contains="" />
         <constraint name="y" minCount="0" maxCount="2147483647" within="" contains="" />
       </searchConfiguration>
+      <searchConfiguration name="Assign an ExecutorService instance to an ExecutorService Variable, not an Executor Variable" text="$x$ = $y$;" recursive="true" caseInsensitive="true" type="JAVA" >
```
Comment (clintropolis) [inline .idea/inspectionProfiles/Druid.xml:375]: nit: `Variable` -> `variable`

File: .idea/inspectionProfiles/Druid.xml [hunk 2]
```diff
@@ -372,6 +372,16 @@
...
+        <constraint name="__context__" within="" contains="" />
+        <constraint name="x" nameOfExprType="java\.util\.concurrent\.Executor" within="" contains="" />
+        <constraint name="y" nameOfExprType="java\.util\.concurrent\.ExecutorService" exprTypeWithinHierarchy="true" within="" contains="" />
+      </searchConfiguration>
+      <searchConfiguration name="Intialize an ExecutorService instance to an ExecutorService Variable, not an Executor Variable" text="Executor $x$ = $y$;" recursive="true" caseInsensitive="true" type="JAVA" >
```
Comment (clintropolis) [inline .idea/inspectionProfiles/Druid.xml:380]: nit: same comment on variable

**Review by clintropolis — COMMENTED** [review]
:+1:

File: .idea/inspectionProfiles/Druid.xml (hunk 1 above)
Comment (shankeerthan-kasilingam) [reply .idea/inspectionProfiles/Druid.xml]: Done

File: .idea/inspectionProfiles/Druid.xml (hunk 2 above)
Comment (shankeerthan-kasilingam) [reply .idea/inspectionProfiles/Druid.xml]: Done

**Review by clintropolis — APPROVED** [review]

**Comment by clintropolis** [comment]
CI is failing due to coverage check, which can be ignored because this change is cosmetic

**Pull request merged**